### ISSUE TITLE
AMD Adrenalin Software reports incorrect display resolution

### ISSUE THREAD

**Issue by jnahumphreys** [itsmikethetech/Virtual-Display-Driver#112]
I'm using the Vitual Display Driver as part of a headless game streming host (with Sunshine and Moonlight), streaming to an iPad Pro 12.9 inch at native resolution - 60fps.

I am looking to leveage global FSR scaling for one particular game via AMD Adrenalin software. To enable FSR you must set the in-game resolution to a lower value than native resolution, FSR wil then automatically upscale. 

The issue I'm facing is the AMD Adrenalin Software is identifying my monior resolution as `640x480`, which is incorrect (windows display settings reports the correct resolution). With AMD Adrenalin unable to identify the correct resolution, I'm unable to enable the global FSR feature.

I suspect this is due to the exclusive use of the IDD driver (i.e no physical monitor connected to the host) perhaps?

Does anyone know of a fix?

**Comment by itsmikethetech** [comment]
Can you try removing all unneeded resolutions from options.txt so that it defaults to the correct resolution? Unless of course you use more than one. Be sure to leave the space at the end of the options.txt file as well.

We have a solution in the source (and upcoming built driver) that allows directly attaching to a GPU.

**Comment by jnahumphreys** [comment]
@itsmikethetech - thanks for the reply and appologise for the delay in mine. 

I've edited the `options.txt` file to contain only one resolution `2732 x 1960 @60Hz`. Unfortunately AMD Adrenaline still reports a display resolution of `640 x 480`.

> We have a solution in the source (and upcoming built driver) that allows directly attaching to a GPU

When are you expecting to do a release with this fix?

**Comment by bud3699** [comment]
Whenever I'm finished with code optimisations hopefully!

**Comment by radugrecu97** [comment]
@itsmikethetech Directly attaching to GPU sounds great. By source you mean on one of the available branches or is it private?
I would love to try it out, if possible 😄

**Comment by itsmikethetech** [comment]
Beta HDR version with GPU switching out now. First time public, so there could be bugs. https://github.com/itsmikethetech/Virtual-Display-Driver/releases/tag/24.7.18HDR

**Comment by jnahumphreys** [comment]
> We have a solution in the source (and upcoming built driver) that allows directly attaching to a GPU.

@itsmikethetech - do we belive this will fix the issue with AMD Adrenaline not detecting the correct resolution of the virtual display?

Keen to leverage system wide FSR, but will likely hold off 'till a stable release as this is my primary gaming rig 

**Comment by radugrecu97** [comment]
@jnahumphreys Did you actually get VDD listed as a display in Adrenaline, or are you getting the resolution value from somewhere else?

Adrenaline only lists the physical monitors on my end. Never VDD, so I'm missing Monitor features.

**Comment by simonetagliaferri** [comment]
Is there something to do to even have the virtual display show up in Adrenalin? I have the beta release with the xml config files, I manually set the gpu in the config, I had to update to Windows 11 24H2 to even have hdr working.

**Comment by sofmeright** [comment]
Just dropping in to leave my two sense. Unfortunate if you can not get FSR working. I have not been able to get the display to detect in AMD Adrenaline, I think that is by design as it is a Virtual Display Adapter; by design it is not attached to the GPU so it is not available to set custom resolutions ETC within Adrenaline. 

I particularly am entering the chat to perhaps suggest a solution in relation to a separate issue I encountered and was able to resolve after investigating. When playing most games they would have no issue rendering at the resolution I have set in windows and the game itself. Immortals Fenyx Rising however, did not cooperate. It would have black bars at the top and bottom of the screen and a weird effect on the sides. In the Video settings it showed my native resolution as 1024 x 768 and aspect ratio as 4:3 in troubleshooting I suspected this was my #1 clue to resolving the issue. I determined that in the options.txt 1024 x 768 was the first resolution entry as I had it sorted by resolution. I shifted my preferred resolution to be the first entry and reloaded the Virtual Display Driver and sure enough my scaling was resolved. If the offending game intended to use FSR on has similar backend it may resolve your issue to try this. I have no issues with HDR and I did not have to use the Beta driver with adapter.txt, I tried it but after finding the solution I reverted and still have no issue.

I am also including my workstation guide you can review briefly that includes some additional info that explains everything I have done. http://sh.uni2.cc/28JLJ

**Comment by bud3699** [comment]
Yes, the since the display doesn't take a physical port, applications which control these will struggle to interface with these virtual displays, and as for the order of displays, yes that's correct, the "recommended" resolution will always be the first resolution inputted
